vortex-expr/src/get_item.rs

```diff
@@ -0,0 +1,128 @@
+use std::any::Any;
+use std::fmt::{Debug, Display, Formatter};
+use std::sync::Arc;
+
+use vortex_array::ArrayData;
+use vortex_dtype::Field;
+use vortex_error::{vortex_err, VortexResult};
+
+use crate::{ExprRef, VortexExpr};
+
+#[derive(Debug, Clone, Eq)]
+pub struct GetItem {
+    field: Field,
+    child: ExprRef,
+}
+
+impl GetItem {
+    pub fn new_expr(field: impl Into<Field>, child: ExprRef) -> ExprRef {
+        Arc::new(Self {
+            field: field.into(),
+            child,
+        })
+    }
+
+    pub fn field(&self) -> &Field {
+        &self.field
+    }
+
+    pub fn child(&self) -> &ExprRef {
+        &self.child
+    }
+}
+
+pub fn get_item(field: impl Into<Field>, child: ExprRef) -> ExprRef {
+    GetItem::new_expr(field, child)
+}
+
+impl PartialEq<dyn Any> for GetItem {
+    fn eq(&self, other: &dyn Any) -> bool {
+        other
+            .downcast_ref::<GetItem>()
+            .map(|item| self.field == item.field && self.child.eq(&item.child))
+            .unwrap_or(false)
+    }
+}
+
+impl Display for GetItem {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}.{}", self.child, self.field)
+    }
+}
+
+impl VortexExpr for GetItem {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn evaluate(&self, batch: &ArrayData) -> VortexResult<ArrayData> {
+        let child = self.child.evaluate(batch)?;
+        let st = child
+            .as_struct_array()
+            .ok_or_else(|| vortex_err!("GetItem: child array into struct"))?;
+
+        match &self.field {
+            Field::Name(name) => st.field_by_name(name.as_ref()),
+            Field::Index(idx) => st.field(*idx),
+        }
+        .ok_or_else(|| vortex_err!("Field {} not found", self.field))
```

Comment:
Note to ourselves: this ignores the validity of the struct array. A pervasive problem, but something we need to deal with.

Reply:
Do we have the apply nulls compute function I think we should use that?

Reply:
Or provide field/field by name, with validity?